### DISCUSSION TITLE
Examples for logging

### DISCUSSION THREAD

**Discussion by smelgin-absa** [codefriar/ApexKit#170] [Q&A] [answered]
Hi @codefriar Kevin! Thanks for this awesome library.

**Comment by codefriar** [comment, marked as answer]
@smelgin-absa - Thanks for reaching out. 

I've tried to make the Log class as easy-to-use as possible. You'll need to make sure the full set of metadata components is installed. -  there's a trigger, a custom platform event and the classes in the classes/log folder. 

I've tried to provide examples in the classes/log/examples/readme.md file, but in case those don't make sense...

In general the first step is to get an instance of the logger by calling something like this:

`Log log = Log.get();` - this gives you access to the singleton logger. 

From there, there are you can call any of the `.add()` or `.publish()` methods. There are plenty of overrides so basically anything you throw at either one will be accepted. The key difference between these two sets of methods is whether or not they immediately emit the platform event. if you call the `.add()` methods, it will result in a queue of events being created and held in the singleton's memory. If you call `.publish()` methods it will add that log entry, and then immediately publish any log entries in the queue. 

Use `.add()` if you're going to rapid-fire add multiple log lines within say, the same method. Otherwise, I generally just call `log.publish('string');` or some variant there of. 

Hope this helps.
